src/pages/user_profile/UserProfile.js

```diff
@@ -27,6 +30,24 @@ export default function UserProfile(props) {
 
   const [notificationData, setNotificationData] = useState({});
 
+  const updateRef = ({key, value}) => {
+    setName(value);
+    let newUser = cloneDeep(user);
+    newUser.displayName = value;
+    setUser(newUser);
```

Comment:
When this is called, it changes the user in state. That triggers a re-render of ResearcherStatus. That triggers a refresh to `User.getSOsForCurrentUser()` in useEffect. The end result of this is that every keystroke triggers an unnecessary call to get signing officials for the current user. Digging in to see if I can find a cleaner approach.

Reply:
OK, here's my solution to this problem.
1. Create a new state variable that will not trigger the researcher status re-render: `const [updatedUser, setUpdatedUser] = useState({});`
2. Update that object instead of the user object when calling `updateRef`: 
```
  const updateRef = ({key, value}) => {
    setName(value);
    let newUser = cloneDeep(user);
    newUser.displayName = value;
    setUpdatedUser(newUser);
  };
```
3. In `updateName`, use that variable to make the API call. Also, only call the storage if the response is a sucess:
```
  const updateName = () => {
    User.update(updatedUser, updatedUser.userId).then((response) => {
      setUserRoleStatuses(response, Storage);
      Notifications.showSuccess({ text: 'Name updated successfully!' });
    }, () => {
      Notifications.showError({ text: 'Some errors occurred, the user\'s name was not updated.' });
    });
  };
```

There might be some additional work to ensure that the save button works in the case that the user hasn't changed their name, but this approach at least solves the nested component re-render problem.

Reply:
I added some logic to ensure we aren't getting an undefined API call if the user hasn't changed their name here.